pyeudiw/satosa/backend.py

```diff
@@ -0,0 +1,195 @@
+import logging
+import base64
+
+from urllib.parse import urlencode, quote_plus
+from satosa.response import Response
+from satosa.backends.base import BackendModule
+
+from pyeudiw.tools.jwk import JWK
+from pyeudiw.tools.jwt import JWSHelper
+
+logger = logging.getLogger(__name__)
+
+
+class OpenID4VPBackend(BackendModule):
+    """
+    A backend module (acting as a OpenID4VP SP).
+    """
+
+    def __init__(self, auth_callback_func, internal_attributes, config, base_url, name):
+        super().__init__(auth_callback_func, internal_attributes, base_url, name)
+
+        self.entity_configuration_url = config['entity_configuration_endpoint']
+        self.pre_request_url = config['pre_request_endpoint']
+        self.redirect_url = config['redirect_endpoint']
+        self.request_url = config['request_endpoint']
+        self.error_url = config['error_url']
+
+        self.client_id = config['wallet_relying_party']['client_id']
+        self.complete_redirect_url = config['wallet_relying_party']['redirect_uris'][0]
+        self.complete_request_url = config['wallet_relying_party']['request_uris'][0]
+
+        self.qr_settings = config['qr_code_settings']
+        self.config = config
+
+    def register_endpoints(self):
+        """
+        Creates a list of all the endpoints this backend module needs to listen to. In this case
+        it's the authentication response from the underlying OP that is redirected from the OP to
+        the proxy.
+        :rtype: Sequence[(str, Callable[[satosa.context.Context], satosa.response.Response]]
+        :return: A list that can be used to map the request to SATOSA to this endpoint.
+        """
+        url_map = []
+        url_map.append(
+            (f"^{self.entity_configuration_url.lstrip('/')}$", self.entity_configuration))
+        url_map.append(
+            (f"^{self.pre_request_url.lstrip('/')}$", self.pre_request_endpoint))
+        url_map.append(
+            (f"^{self.redirect_url.lstrip('/')}$", self.redirect_endpoint))
+        url_map.append(
+            (f"^{self.request_url.lstrip('/')}$", self.request_endpoint))
+        return url_map
+
+    def entity_configuration(self, context, *args):
+        jwk = JWK()
+
+        data = {
+            "exp": 1649590602,
+            "iat": 1649417862,
+            "iss": "https://rp.example.it",
+            "sub": "https://rp.example.it",
+            "jwks": {
+                "keys": [jwk.export_public()]
+            },
+            "metadata": {
+                "wallet_relying_party": self.config['wallet_relying_party']
+            }
+        }
+
+        jwshelper = JWSHelper(jwk)
+
+        return Response(
+            jwshelper.sign(
+                plain_dict=data,
+                protected={
+                    "alg": "RS256",
+                    "kid": "2HnoFS3YnC9tjiCaivhWLVUJ3AxwGGz_98uRFaqMEEs",
+                    "typ": "entity-statement+jwt"
+                }
+            ),
+            status="200 OK"
```

Comment:
sometimes you use int 200, other times a string

please use the int in the the status response